File: leetcode/python/2441_largest_positive_integer_that_exists_negative.py
import heapq


class Solution:
    def findMaxK(self, nums: list[int]) -> int:
        numSet = set(nums)
        sortedNums = sorted(nums, reverse=True)
        for n in sortedNums:
            if -n in numSet:
                return n
        return -1

s = Solution()
print(s.findMaxK([-10,8,6,7,-2,-3]))
print(s.findMaxK([-1,2,-3,3]))
print(s.findMaxK([-1,10,6,7,-7,1]))

File: leetcode/rust/src/2441_largest_positive_integer_that_exists_negative.rs
use std::collections::HashSet;

impl Solution {
    pub fn find_max_k(nums: Vec<i32>) -> i32 {
        let mut nums_set = HashSet::new();
        let mut res = -1;
        for num in nums {
            nums_set.insert(num);
            if nums_set.contains(&-num) {
                res = res.max(num.abs());
            }
        }
        res
    }
}
